### PR TITLE
chore: update changelog for v2.4.0

### PR DESCRIPTION
**Pull request by devbww**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10158)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -131,6 +131,65 @@ In addition, these existing libraries are now GA:
...
+- doc: add endpoint override snippets to generated libs ([#10129](https://github.com/googleapis/google-cloud-cpp/pull/10129))
+
+### [Pub/Sub](/google/cloud/pubsub/README.md)
+
+- doc(pubsub): samples for endpoint and auth ([#10136](https://github.com/googleapis/google-cloud-cpp/pull/10136))
+- feat(pubsub): install pubsub_mocks pkg ([#10008](https://github.com/googleapis/google-cloud-cpp/pull/10008))
+- feat(pubsub): implement blocking publisher ([#10055](https://github.com/googleapis/google-cloud-cpp/pull/10055))
+- feat(pubsub): implement per-call options for `Subscriber` ([#10043](https://github.com/googleapis/google-cloud-cpp/pull/10043))
+- fix(pubsub): limit `ModifyAckDeadlineRequest` size ([#10032](https://github.com/googleapis/google-cloud-cpp/pull/10032))
+- fix(pubsub): race condition capturing batching timer ([#9996](https://github.com/googleapis/google-cloud-cpp/pull/9996))
```
Comment (coryan) [inline CHANGELOG.md:155]: This is a fix for the next fix, you can collapse the two.

File: CHANGELOG.md [hunk 2]
```diff
@@ -131,6 +131,65 @@ In addition, these existing libraries are now GA:
...
+- feat(spanner): support for the PG.JSONB data type ([#10098](https://github.com/googleapis/google-cloud-cpp/pull/10098))
+
+### [Storage](/google/cloud/storage/README.md)
+
+- doc(storage): common initialization examples ([#10107](https://github.com/googleapis/google-cloud-cpp/pull/10107))
+- fix(rest): too many debug headers ([#10054](https://github.com/googleapis/google-cloud-cpp/pull/10054))
+- fix(rest): return complete payloads for errors ([#10051](https://github.com/googleapis/google-cloud-cpp/pull/10051))
+- feat(storage): support `Autoclass` feature ([#10003](https://github.com/googleapis/google-cloud-cpp/pull/10003))
+- feat(storage): faster `InsertObject()` uploads ([#9997](https://github.com/googleapis/google-cloud-cpp/pull/9997))
+- feat(GCS+gRPC): use alternative names in ACLs ([#9992](https://github.com/googleapis/google-cloud-cpp/pull/9992))
```
Comment (coryan) [inline CHANGELOG.md:171]: We are still skipping GCS+gRPC changes.

**Review by coryan — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -131,6 +131,65 @@ In addition, these existing libraries are now GA:
...
+
+- feat(generator): scaffold supports auth samples ([#10146](https://github.com/googleapis/google-cloud-cpp/pull/10146))
+- feat(generator): generate authentication example ([#10138](https://github.com/googleapis/google-cloud-cpp/pull/10138))
+- doc: workaround Doxygen formatting quirk ([#10137](https://github.com/googleapis/google-cloud-cpp/pull/10137))
+- doc: add endpoint override snippets to generated libs ([#10129](https://github.com/googleapis/google-cloud-cpp/pull/10129))
+- feat(generator): add markers during scaffold generation ([#10127](https://github.com/googleapis/google-cloud-cpp/pull/10127))
+- feat(generator): includes samples in DOXYGEN_EXAMPLE_PATH ([#10126](https://github.com/googleapis/google-cloud-cpp/pull/10126))
+- feat(generator): build scripts for samples ([#10125](https://github.com/googleapis/google-cloud-cpp/pull/10125))
+- feat(generator): generate simple samples for `*Client` ([#10118](https://github.com/googleapis/google-cloud-cpp/pull/10118))
+- doc: mocks library / StreamRange ([#10122](https://github.com/googleapis/google-cloud-cpp/pull/10122))
```
Comment (dbolduc) [inline CHANGELOG.md:184]: nit: I think we can delete this one, because it is covered by the change below.

File: CHANGELOG.md (hunk 1 above)
Comment (devbww) [reply CHANGELOG.md]: Done (by deleting the "fix race").

File: CHANGELOG.md (hunk 2 above)
Comment (devbww) [reply CHANGELOG.md]: Removed.

File: CHANGELOG.md [hunk 4]
```diff
@@ -131,6 +131,65 @@ In addition, these existing libraries are now GA:
...
+- feat: add mock library w/ StreamRange ([#9998](https://github.com/googleapis/google-cloud-cpp/pull/9998))
+- doc(common): several fixes from a post-merge review ([#10111](https://github.com/googleapis/google-cloud-cpp/pull/10111))
```
Comment (dbolduc) [inline CHANGELOG.md:184]: I think this can be deleted too.

File: CHANGELOG.md (hunk 3 above)
Comment (devbww) [reply CHANGELOG.md]: Removed.

File: CHANGELOG.md [hunk 5]
```diff
@@ -131,6 +131,65 @@ In addition, these existing libraries are now GA:
...
+- fix(rest): too many debug headers ([#10054](https://github.com/googleapis/google-cloud-cpp/pull/10054))
+- fix(rest): return complete payloads for errors ([#10051](https://github.com/googleapis/google-cloud-cpp/pull/10051))
+- feat(storage): support `Autoclass` feature ([#10003](https://github.com/googleapis/google-cloud-cpp/pull/10003))
+- feat(storage): faster `InsertObject()` uploads ([#9997](https://github.com/googleapis/google-cloud-cpp/pull/9997))
+- feat(GCS+gRPC): use alternative names in ACLs ([#9992](https://github.com/googleapis/google-cloud-cpp/pull/9992))
+- fix(storage): respect MIME message boundary size limits ([#9965](https://github.com/googleapis/google-cloud-cpp/pull/9965))
+
+### [Common Libraries](/google/cloud/README.md)
+
+- feat(generator): scaffold supports auth samples ([#10146](https://github.com/googleapis/google-cloud-cpp/pull/10146))
```
[the remark this answers is not in the document]
Comment (devbww) [reply CHANGELOG.md]: Removed.

File: CHANGELOG.md [hunk 6]
```diff
@@ -131,6 +131,65 @@ In addition, these existing libraries are now GA:
...
+- feat(generator): generate authentication example ([#10138](https://github.com/googleapis/google-cloud-cpp/pull/10138))
+- doc: workaround Doxygen formatting quirk ([#10137](https://github.com/googleapis/google-cloud-cpp/pull/10137))
+- doc: add endpoint override snippets to generated libs ([#10129](https://github.com/googleapis/google-cloud-cpp/pull/10129))
+- feat(generator): add markers during scaffold generation ([#10127](https://github.com/googleapis/google-cloud-cpp/pull/10127))
+- feat(generator): includes samples in DOXYGEN_EXAMPLE_PATH ([#10126](https://github.com/googleapis/google-cloud-cpp/pull/10126))
+- feat(generator): build scripts for samples ([#10125](https://github.com/googleapis/google-cloud-cpp/pull/10125))
```
[the remark this answers is not in the document]
Comment (devbww) [reply CHANGELOG.md]: Removed x3.

File: CHANGELOG.md (hunk 4 above)
Comment (devbww) [reply CHANGELOG.md]: Removed.

**Pull request merged**